akka-stream-tests/src/test/scala/akka/stream/io/InputStreamSinkSpec.scala

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.io
+
+import java.io.{ IOException, InputStream }
+import java.util.concurrent.{ BlockingQueue, TimeoutException }
+
+import akka.actor.{ ActorSystem, Deploy, Props }
+import akka.stream._
+import akka.stream.actor.ActorSubscriber.OnSubscribe
+import akka.stream.actor.ActorSubscriberMessage.{ OnComplete, OnError, OnNext }
+import akka.stream.impl.Stages.DefaultAttributes
+import akka.stream.impl.StreamSupervisor.Children
+import akka.stream.impl.io.{ InputStreamSink, InputStreamSubscriber }
+import akka.stream.impl.{ ActorMaterializerImpl, SinkModule, StreamSupervisor }
+import akka.stream.scaladsl.{ Keep, Sink }
+import akka.stream.testkit.AkkaSpec
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.TestSource
+import akka.testkit.TestProbe
+import akka.util.ByteString
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future, Promise }
+import scala.util.control.NoStackTrace
+
+class InputStreamSinkSpec extends AkkaSpec(UnboundedMailboxConfig) {
+  import system.dispatcher
+
+  val settings = ActorMaterializerSettings(system).withDispatcher("akka.actor.default-dispatcher")
+  implicit val materializer = ActorMaterializer(settings)
+
+  val timeout = 300.milliseconds
+  val byteString = ByteString("abc")
+  val byteArray = "abc".toArray
```

Comment:
java.lang.String does not have a toArray method, so I assume it is added by Scala, and I don't know what char encoding it uses. I'd recommend doing: ByteString.fromString("abc", "UTF-8").toArray